### ISSUE TITLE
MSBuild props/targets not included in designtime builds during project creation

### ISSUE THREAD

**Issue by mgoertz-msft** [dotnet/project-system#7854]
## Visual Studio Version
17.1 Preview 3

## Summary
Looks like NuGet Restore has not run or completed yet during evaluation of DT builds causing the props/targets in the project's obj folder generated during restore to not be included in any DT builds:
`NoImport: $(MSBuildProjectExtensionsPath)$(MSBuildProjectFile).*.targets at (128;3) false condition; ('$(ImportProjectExtensionTargets)' == 'true' and exists('$(MSBuildProjectExtensionsPath)') was evaluated as 'true' == 'true' and exists('C:\Users\mgoertz\source\repos\MauiApp17\MauiApp17\obj\')).`

## Steps to Reproduce

1. Install VS 17.1 Preview 3 with Xamarin Mobile workload and optional MAUI selected
2. Add these required to your %APPDATA%\NuGet\NuGet.config:
dotnet nuget add source "https://api.nuget.org/v3/index.json" --name "nuget" 
dotnet nuget add source "https://aka.ms/dotnet6/nuget/index.json" --name "dotnet6" 
3. Launch VS
4. Create new MAUI app project

## Expected Behavior
No errors

## Actual Behavior
C# IntelliSense errors:
1. The name 'InitializeComponent' does not exist in the current context
2. The name 'CounterLabel' does not exist in the current context
3. The name 'CounterLabel' does not exist in the current context

## User Impact
Project builds and runs but IntelliSense errors don't go away until solution close and reopen.

**Comment by tmeschter** [comment]
I don't believe there is actual work for us here; next step is to review the related Teams conversation and figure out the next steps.

**Comment by mgoertz-msft** [comment]
MAUI has worked around this issue by moving their source generator and related tasks out of NuGet. However, the general limitation remains - no targets/props from NuGet packages are part of DT builds during project creation because the necessary targets/props responsible for pulling those in are generated during NuGet restore.

**Comment by drewnoakes** [comment]
We need to run a DTB in order to work out what to restore. So there's a chicken and egg situation.

**Comment by mgoertz-msft** [comment]
I guess the restore would need to trigger another DTB. Someone said that was supposed to happen automatically, but it didn't look like that was working correctly. While there were 3 DTBs per TFM during MAUI project creation none of them looked like they were including the targets from the MAUI package.

Anyway, perhaps this is now a much lower priority since it's no longer needed for MAUI.
